aircan/dags/ng_data_dictionary_reporting.py

```diff
@@ -0,0 +1,148 @@
+import logging
+import datetime
+import time
+from pprint import pprint
+
+from airflow import DAG
+from airflow.exceptions import AirflowException
+from airflow.models import Variable
+import requests
+from urllib.parse import urljoin
+from airflow.operators.python_operator import PythonOperator
+from airflow.utils.dates import days_ago
+from airflow.operators.email_operator import EmailOperator
+from sendgrid import SendGridAPIClient
+from sendgrid.helpers.mail import Mail
+from string import Template
+from time import sleep
+
+CKAN_URL =  Variable.get("NG_CKAN_URL")
+SENDGRID_KEY =  Variable.get("NG_SENDGRID_API_KEY")
+MAIL_FROM = Variable.get("NG_MAIL_FROM")
+MAIL_TO = Variable.get("NG_MAIL_TO")
+
+args = {
+    'start_date': days_ago(0)
+}
+
+dag = DAG(
+    dag_id='ng_data_example_logs',
+    default_args=args,
+    schedule_interval='0 0 * * *'
+)
+
+def get_all_datastore_tables():
+    try:
+        logging.info("Retriving all datastore tables.")
+        response = requests.get(
+            urljoin(CKAN_URL, '/api/3/action/datastore_search?resource_id=_table_metadata')
+        )
+        response.raise_for_status()
+        if response.status_code == 200:
+           return [resource['name'] for resource in response.json()['result']['records']]
+            
+    except requests.exceptions.HTTPError as e:
+            return e.response.text 
+
+
+def get_data_dictionary(res_id):
+    try:
+        logging.info("Retriving fields infomation for {res_id}".format(res_id=res_id))
+        response = requests.get(
+            urljoin(CKAN_URL, 
+                    '/api/3/action/datastore_search?resource_id={0}&limit=0'.format(res_id))
```

Comment:
@sagargg what if we use `datastore_info` API?

Also, can we add user agent here so that we can filter out these calls in analytics data?